petstore/src/main/scala/io/finch/petstore/package.scala

```diff
@@ -0,0 +1,76 @@
+package io.finch
+
+import io.finch.request.ToRequest
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.collection.JavaConverters._
+
+import com.twitter.util.Future
+
+/**
+ * The demo project shows the usage of Finch's basic blocks for building a purely functional REST API backend
+ * emulating a set of services working with pet stores, pets, and users (who purchase the pets).
+ *
+ * The following packages represent the backend:
+ *
+ * - [[petstore.model]] - domain model classes: Pet, Store, User
+ * - [[petstore.reader]] - [[io.finch.request.RequestReader]]s for models
+ * - [[petstore.service]] - the application services
+ * - [[petstore.endpoint]] - [[io.finch.route.Router]]s for services (endpoints)
+ */
+
+package petstoreDemo {
+
+import com.twitter.util.Future
+import io.finch._
+import io.finch.demo.model
+import io.finch.demo.model.User
+import io.finch.request.ToRequest
+import model._
+
+// A custom request type that wraps an `HttpRequest`.
+// We prefer composition over inheritance.
+case class AuthRequest(http: HttpRequest)
```

Comment:
As discussed yesterday, we can scrap this type since we're going to start without authentication (since eventually it'll look different from this, anyway).